fe/fe-core/src/main/java/org/apache/doris/statistics/IcebergAnalysisJob.java

```diff
@@ -0,0 +1,83 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.statistics;
+
+import org.apache.doris.persist.AnalysisJobScheduler;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.types.Types;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class IcebergAnalysisJob extends HMSAnalysisJob {
+
+    public IcebergAnalysisJob(AnalysisJobScheduler analysisJobScheduler, AnalysisJobInfo info) {
+        super(analysisJobScheduler, info);
+    }
+
+    @Override
+    protected void getPartitionColumnStatsByMeta() {
+        Table icebergTable = getIcebergTable();
+        TableScan tableScan = icebergTable.newScan().includeColumnStats();
+        HMSStatistic stats = new HMSStatistic();
+        for (FileScanTask task : tableScan.planFiles()) {
+            processDataFile(task.file(), task.spec(), stats);
+        }
+    }
+
+    private Table getIcebergTable() {
+        org.apache.iceberg.hive.HiveCatalog hiveCatalog = new org.apache.iceberg.hive.HiveCatalog();
```

Comment:
Why not get table from HMSExternalTable directly?

Reply:
This function is to get org.apache.iceberg.Table.
getTable in HMSExternalTable is to get org.apache.hadoop.hive.metastore.api.Table.
I think we can keep the code here.